backend/src/transaction_event/protocol_level_tokens.rs

```diff
@@ -613,51 +615,249 @@ impl PreparedTokenUpdate {
                 event_type,
                 token_module_type,
                 token_index,
-                token_event,
-                event_timestamp,
-                amount_value,
-                amount_decimals
+                token_event
             )
             VALUES (
                 (SELECT COALESCE(MAX(id) + 1, 0) FROM plt_events),
                 $1,
                  $2,
                  $3,
                 (SELECT index FROM plt_tokens WHERE token_id = $4),
-                $5,
-                (SELECT slot_time from blocks WHERE height = (SELECT block_height FROM \
-             transactions WHERE index = $1)), 
-                $6,
-                $7)
+                $5
+              
+                )
             ",
             transaction_index,
             self.event_type as TokenUpdateEventType,
             self.token_module_type as Option<TokenUpdateModuleType>,
             self.token_id,
-            token_event,
-            self.amount_value,
-            self.amount_decimals
+            token_event
         )
         .execute(tx.as_mut())
         .await?;
 
+        self.update_metrics_plt_cumulative_event_count(tx, slot_time).await?;
+
         match self.event_type {
             TokenUpdateEventType::Mint => {
                 if let Some(ref target) = self.target {
+                    let account_exists = self.check_if_account_exists(tx, target).await?;
                     self.update_total_minted(tx).await?;
                     self.update_account_balance(tx, target, &self.plt_amount_change).await?;
+                    if !account_exists {
+                        self.update_metrics_plt_unique_account_count_delta(tx, slot_time, 1)
+                            .await?;
+                    }
                 }
             }
             TokenUpdateEventType::Burn => {
                 if let Some(ref target) = self.target {
                     self.update_total_burned(tx).await?;
-                    self.update_account_balance(tx, target, &(-&self.plt_amount_change)).await?;
+                    let final_balance = self
```

Comment:
It needs to be the `final_balance` across all PLTs, isn't it? Currently, it is the `final_balance` for the given `token_id`. We only want to increment the counter if the account has no PLTs at all.